### PR TITLE
Whitelist allowed origins running js version of the site

### PR DESCRIPTION
**Pull request by harrison**
Probably a terrible idea.. need think this through.

### CODE REVIEW

**Comment by harrison** [comment]
Fixes https://github.com/gocardless/splash-pages/issues/143

**Comment by english** [comment]
What's the state of this @harrison?

**Comment by harrison** [comment]
Good question, what do you reckon about the approach?:) feels icky..

**Comment by english** [comment]
Yeah, I see what you mean. There's not many alternatives.

What about just checking whether we're in an iframe? Something along the lines of: http://stackoverflow.com/questions/326069/how-to-identify-if-a-webpage-is-being-loaded-inside-an-iframe-or-directly-into-t

**Comment by Ken-Lo** [comment]
@english I'm not sure it's technically an iframe, Google cache simply outputs the HTML of our page and  then calls cross-domain Javascript from http://webcache.googleusercontent.com/. So not sure the code there will work.

All-in-all, it's not super urgent to fix this issue especially if it's icky or has the chance of breaking things. Blacklisting googleusercontent.com, web.archive.org, etc. is probably another way to do this - equally as icky but probably has less chance of breaking things.

**Comment by english** [comment]
Thanks @Ken-Lo, I didn't realise that. In that case, then I guess this either blacklisting or whitelisting would be fine.

**Comment by harrison** [comment]
hm yeah might do a blacklist instead

**Comment by harrison** [comment]
@english wanna give this a last check? webarchive doesn't run any js so no need to blacklist

File: config/blacklisted-origins.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const blacklistedHosts = [
+  'webcache.googleusercontent.com',
+];
+
+export function isHostBlacklisted(host) {
+  return blacklistedHosts.some((blanklistedHost) => blanklistedHost === host);
```
Comment (english) [inline config/blacklisted-origins.js]: I'm curious if there's any reason you didn't use `_.contains` or `indexOf`?

**Comment by english** [comment]
Looks good to me :+1:

File: config/blacklisted-origins.js (hunk 1 above)
Comment (harrison) [inline config/blacklisted-origins.js]: oh yeah good point, was doing a regex before, so needed `some`

**Pull request merged**